api/queries/trending-repos/template.sql

```diff
@@ -0,0 +1,165 @@
+                  ), 5
+              ), 2
+          )
+      ) AS score
+  FROM github_events ge
+  WHERE
+      -- Notice: In the GitHub events, WatchEvent means star, not watch.
+      type = 'WatchEvent'
+      AND (ge.created_at >= DATE_SUB(NOW(), INTERVAL 1 MONTH) AND ge.created_at <= NOW())
+  GROUP BY ge.repo_id
+  -- Exclude code repositories that use the same user to duplicate stars.
+  HAVING actors > 0.9 * total
+), forks AS (
+  SELECT
+      ge.repo_id AS repo_id,
+      COUNT(1) AS total,
+      COUNT(DISTINCT actor_id) AS actors,
+      -- Calculate the score of each fork according to the time of the fork, the closer to the 
+      -- current time, the higher the score got, the score range is between 1-4. Then sum the
+      -- scores of all forks to get the total score obtained from the forks for the repository.
+      SUM(
+          GREATEST (
+              LEAST (
+                  (
+                      (
+                          TIMESTAMPDIFF(SECOND, DATE_SUB(NOW(), INTERVAL 1 MONTH), ge.created_at) / 
+                          TIMESTAMPDIFF(SECOND, DATE_SUB(NOW(), INTERVAL 1 MONTH), NOW())
+                      ) * (4 - 1)
+                  ), 4
+              ), 1
+          )
+      ) AS score
+  FROM github_events ge
+  WHERE
+      type = 'ForkEvent'
+      AND (ge.created_at >= DATE_SUB(NOW(), INTERVAL 1 MONTH) AND ge.created_at <= NOW())
+  GROUP BY ge.repo_id
+  -- Exclude code repositories that use the same user to duplicate froks.
+  HAVING actors > 0.9 * total
+), topRepos AS (
+    SELECT
+        r.repo_id,
+        r.repo_name,
+        r.primary_language,
+        r.description,
+        IFNULL(s.total, 0) AS stars_inc,
+        IFNULL(f.total, 0) AS forks_inc,
+        -- Calculate the composite score for the repository.
+        SUM(
+            IFNULL(s.score, 0) + 
+            IFNULL(f.score, 0) +
+            -- Give the new repository a higher score base.
+            ABS(1 /  (1 + TIMESTAMPDIFF(YEAR, r.created_at, NOW()))) * 200
+        ) AS total_score
+    FROM github_repos r
+        JOIN stars s ON r.repo_id = s.repo_id
+        LEFT JOIN forks f ON r.repo_id = f.repo_id
+    WHERE
+        -- Filter rule: The repository must have at least 5 stars.
+        stars > 5
+        -- Filter rule: The repository must have at least 5 forks.
+        AND forks > 5
+        -- Filter rule: The repository must have pushed new code within the last three months.
+        AND pushed_at > DATE_SUB(NOW(), INTERVAL 3 MONTH)
+        -- Filter rule: Exclude some malicious new repositories.
+        AND created_at < DATE_SUB(NOW(), INTERVAL 1 DAY)
+        -- Filter rule: There should be no uncivilized words in the name of the repository.
+        AND LOWER(repo_name) NOT LIKE '%fuck%'
+        -- Filter by repository language.
+        AND primary_language = 'Java'
+        AND is_deleted = 0
+    GROUP BY r.repo_id
+    ORDER BY total_score DESC
+    LIMIT 100
+), pull_requests AS (
+  SELECT
```

Comment:
Did we need [`merge()`](https://docs.pingcap.com/tidb/dev/optimizer-hints#merge)  hint here?

Reply:
No, because there is no predicate that can be pushed in the main query, and the results of the results returned by this query are very small..

Reply:
Got it.